partiql-ast/src/main/java/org/partiql/ast/expr/ExprWindow.java

```diff
@@ -67,13 +67,13 @@ public <R, C> R accept(@NotNull AstVisitor<R, C> visitor, C ctx) {
     @lombok.Builder(builderClassName = "Builder")
     @EqualsAndHashCode(callSuper = false)
     public static class Over extends AstNode {
-        @Nullable
+        @NotNull
         public final List<Expr> partitions;
 
-        @Nullable
+        @NotNull
         public final List<Sort> sorts;
 
-        public Over(@Nullable List<Expr> partitions, @Nullable List<Sort> sorts) {
+        public Over(@NotNull List<Expr> partitions, @NotNull List<Sort> sorts) {
```

Comment:
(self-review) prior modeling was slightly overcomplicated. An empty `partitions` or `sorts` list should represent no `PARTITION BY` or `ORDER BY` specifications.

Reply:
This would make a good doc comment!

Reply:
True. I'll add